modules/oidc/core/src/main/scala/ch/epfl/bluebrain/nexus/iam/oidc/config/OidcProviderConfig.scala

```diff
@@ -0,0 +1,65 @@
+package ch.epfl.bluebrain.nexus.iam.oidc.config
+
+import java.nio.charset.StandardCharsets
+
+import akka.actor.ActorSystem
+import akka.http.scaladsl.model.{HttpRequest, Uri}
+import akka.http.scaladsl.model.Uri.ParsingMode
+import akka.stream.Materializer
+import cats.syntax.either._
+import ch.epfl.bluebrain.nexus.commons.http.HttpClient
+import ch.epfl.bluebrain.nexus.commons.http.HttpClient._
+import io.circe.{Decoder, DecodingFailure, HCursor}
+
+import scala.concurrent.Future
+import scala.util.Try
+
+/**
+  * Configuration of an OIDC provider derived from the discovery uri.
+  *
+  * @param authorization the authorization uri
+  * @param token the token uri
+  * @param userInfo the user info uri
+  * @param jwks the jwks uri
+  */
+final case class OidcProviderConfig(authorization: Uri, token: Uri, userInfo: Uri, jwks: Uri)
+
+object OidcProviderConfig {
+
+  /**
+    * Looks up an ''OidcProviderConfig'' at the provided discovery uri.
+    *
+    * @param discoveryUri an OIDC provider discovery uri, typically ending in ''.../.well-known/openid-configuration''
+    */
+  final def apply(discoveryUri: Uri)(implicit
+                                     as: ActorSystem,
+                                     mt: Materializer,
+                                     cl: UntypedHttpClient[Future]): Future[OidcProviderConfig] = {
+    import de.heikoseeberger.akkahttpcirce.FailFastCirceSupport._
+    import as.dispatcher
+    val fixed = implicitly[HttpClient[Future, OidcProviderConfig]]
```

Comment:
what does `fixed` mean in this context? can we rename it to something more meaningful

Reply:
I wanted to express a fixed type client, but I'll rename it.